### PR TITLE
Added tests for src/screens/Users/Users.tsx

### LINKED ISSUE #2376: Write Unittests for `src/screens/Users/Users.tsx`
We will need unittests done for all methods, classes and functions found in this file.

### PR Acceptance Criteria

1. When complete this file must show **100%** coverage when merged into the code base.
1. [The current code coverage for the file can be found here](https://app.codecov.io/gh/PalisadoesFoundation/talawa-admin/tree/develop/?displayType=list)
1. CodeRabbit must approve all your changes to the test file(s) and the file(s) under test
1. The PR will show a report for the code coverage for the file you have added. You can use that as a guide
1. No functionality must be removed to achieve this goal
1. All tests must be valid and truly test the code base

### Other Criteria

1. If this file isn't referenced in the code base then remove it. We don't want tests run on unused code

### Potential internship candidates

Please read this if you are planning to apply for a Palisadoes Foundation internship 
- https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by Aad1tya27**
**What kind of change does this PR introduce?**
Added tests for ```src/screens/Users/Users.tsx``` , and fixed bugs in Users.tsx in sorting and filtering.

**Issue Number:**

Fixes #2376 

**Did you add tests for your changes?**
Yes

**Summary**
To add further tests for the above file.

**Does this PR introduce a breaking change?**
No

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**
Yes

## Summary by CodeRabbit

- **New Features**
	- Enhanced user data management and loading mechanism in the Users component.
	- Improved infinite scrolling with precise control over user fetching.

- **Bug Fixes**
	- Resolved issues related to duplicate user entries and data display.

- **Tests**
	- Updated test suite to cover a broader range of scenarios, including user rendering and search functionality.
	- Enhanced tests for sorting, filtering, and handling user search functionality.
	- Added new mock data to facilitate comprehensive testing.
	- Introduced tests for scenarios with no users or organizations present.

### CODE REVIEW

**Review by palisadoes — CHANGES_REQUESTED** [review]
1. Please convert `src/screens/Users/Users.test.tsx` to use vitest. we are migrating to that platform.
2. The file should be renamed to `src/screens/Users/Users.spec.tsx`
3. Here is an example of a similar conversion 
    1. https://github.com/PalisadoesFoundation/talawa-admin/pull/2742

**Comment by duplixx** [comment]
@Aad1tya27 , are you still working on it? The test code coverage levels for Users.tsx are too low. Please add test cases to bring it to an acceptable range.

**Comment by Aad1tya27** [comment]
> @Aad1tya27 , are you still working on it? The test code coverage levels for Users.tsx are too low. Please add test cases to bring it to an acceptable range.

Yeah I am working on it. I have shifted from jest to vitest. I am facing some errors related to variables not updating which leads to unexpected values.

**Comment by Aad1tya27** [comment]
Actually the time consuming part is fixing the state management in ```Users.tsx``` , because unless I fix it I can't add test cases for them as they would fail anyways.

**Review by palisadoes — CHANGES_REQUESTED** [review]
Please improve the code coverage of the patch so the tests pass, ideally to 100%. The lines that are missing tests can be found by scrolling through this file:

- https://github.com/PalisadoesFoundation/talawa-admin/pull/2745/files#diff-d113732c4f58e36ff2447ed650e53ba5ef39c8db0f03fb7f4e2d263f58b75674

File: src/screens/Users/Users.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,1985 @@
...
+    appUserProfile: {
+      _id: 'user15',
+      adminFor: [],
+      isSuperAdmin: false,
+      createdOrganizations: [],
+      createdEvents: [],
+      eventAdmin: [],
+    },
+  },
+];
```
[the remark this answers is not in the document]
Comment (Aad1tya27) [reply src/screens/Users/Users.spec.tsx]: I have updated my code and got rid of the redundant code.

**Comment by Aad1tya27** [comment]
image
I've added test cases covering almost every line, no line is uncovered, only some of them are partially covered. I hope its acceptable.

**Review by palisadoes — APPROVED** [review]

**Comment by palisadoes** [comment]
You made significant changes. I merged it and opened another issue to get the coverage to 100%

Thanks

**Pull request merged**